### PR TITLE
iOS - Add ability to scroll to caret when typing goes under the keyboard

### PR DESCRIPTION
**Pull request by pinarol**
Fixes: https://github.com/wordpress-mobile/gutenberg-mobile/issues/464

This PR adds ability to follow caret when typing goes under the keyboard. 

caret-follow3

caret-follow4

WPiOS PR: https://github.com/wordpress-mobile/WordPress-iOS/pull/10826

Depends on PRs:
https://github.com/wordpress-mobile/react-native-aztec/pull/112
https://github.com/WordPress/gutenberg/pull/13323
https://github.com/wordpress-mobile/react-native-keyboard-aware-scroll-view/pull/3

**To Test**

Prerequisites

- Run `yarn install` <-- important `yarn start` `yarn ios`
- Might need to remember the scrolling mechanism, reviewers can refer to the diagram from [this PR](https://github.com/wordpress-mobile/react-native-keyboard-aware-scroll-view/pull/2).
 
Test 1 - Paragraph block - Add new text to end
- Add new text at the end of the block until caret goes under the keyboard
- Verify that it automatically refreshes the scroll position to show the caret and also the block toolbar

Note: it is also possible to scroll for only showing the caret without the block toolbar, I wanted to keep this consistent with what happens when we open the keyboard, however, we can still change it for typing case. 

Test 2 - Paragraph block - Add new text in the middle
- Add new text in the middle of the block until caret goes under the keyboard
- Verify that it automatically refreshes the scroll position to show the caret

Test 3 - Repeat tests 1,2 for Heading block

Test 4 - Android

We don't expect any change on Android side.

- Run `yarn android`
- Make sure Android example app opens successfully and you can do basic operations on it

### CODE REVIEW

File: src/block-management/block-manager.js [hunk 1]
```diff
@@ -193,6 +196,19 @@ export class BlockManager extends React.Component<PropsType, StateType> {
 		this.setState( { isKeyboardVisible: false } );
 	}
 
+	onCaretVerticalPositionChange = ( targetId: number, caretY: number, previousCaretY: ?number ) => {
+		if ( ! this.scrollViewRef ) {
+			return;
+		}
+		if ( previousCaretY ) { //if this is not the first tap
+			this.scrollViewRef.props.refreshScrollForField( targetId );
```
[the remark this answers is not in the document]
Comment (pinarol) [reply src/block-management/block-manager.js]: I've moved handling of this event to platform specific files, at least we can make it more obvious this way, let me know what you think

**Comment by pinarol** [comment]
>I did found an issue testing on WPiOS. At the beginning I wasn't sure what was the problem but I think it's related to the iOS version. Testing on iOS 11 (used by default by yarn ios it works fine, but using iOS 12, this happens:

Solved this one, ready for another check @etoledom

**Comment by pinarol** [comment]
@etoledom I think that issue is related with https://github.com/wordpress-mobile/gutenberg-mobile/issues/463 I've linked your comment under that issue. I'll be looking into it when I start to work on that.

Also I have reverted back the scrolling behavior to include the inner toolbar as @iamthomasbishop suggested. @iamthomasbishop you can compare 2 behaviors by trying iOS and Android at the same time, please let me once you have a chance to try it with the new build and if you think that sth needs to be changed don't hesitate to reach out! ( My opinions usually tend to change after trying :) )

**Comment by iamthomasbishop** [comment]
Just got a chance to take a test build for a spin. While I think there are a couple of things that we should do to further improve the interaction, this particular PR seems to solve the originally stated issue. I'm fine with shipping this and making further refinements separately.

Related PR on Android: https://github.com/wordpress-mobile/gutenberg-mobile/issues/427 , as we should make sure Android behavior is matching.

**Pull request merged**